clang-tools-extra/docs/clang-tidy/checks/bugprone/nondeterministic-pointer-usage.rst

```diff
@@ -0,0 +1,31 @@
+.. title:: clang-tidy - bugprone-nondeterministic-pointer-usage
+
+nondeterministic-pointer-usage
+==============================
+
+Finds nondeterministic usages of pointers in unordered containers.
+
+One canonical example is iteration across a container of pointers.
+
+.. code-block:: c++
+
+  {
+    for (auto i : UnorderedPtrSet)
+      f(i);
+  }
+
+Another such example is sorting a container of pointers.
+
+.. code-block:: c++
+
+  {
+    std::sort(VectorOfPtr.begin(), VectorOfPtr.end());
+  }
+
+Iteration of a containers of pointers may present the order of different
+pointers differently across different runs of a program. In some cases this
+may be acceptable behavior, in others this may be unexpected behavior. This
+check is advisory for this reason.
```

Comment:
@vabridgers I really like this **`advisory`** here. Perhaps we could design an `advisory-...` group (akin to `optin.` in CSA?) in Tidy and put checks like this here. "Rules" that are not really part of a publication like CERT or Abseil, but good for use cases where your specific project wants to ensure adherence to some properties, like in this case, making sure to always have a deterministic order.

But perhaps a new group will require more than one candidate.

Reply:
Thanks @whisperity , I'll look for more candidates here, maybe there's at least one more candidate to support the creation of an advisory group.